beluga_amcl/src/amcl_nodelet.cpp

```diff
@@ -424,62 +423,54 @@ void AmclNodelet::laser_callback(ExecutionPolicy&& exec_policy, const sensor_msg
           config_.laser_min_range,
           config_.laser_max_range,
-      }
-              .points_in_cartesian_coordinates() |
-          ranges::views::transform([&](const auto& p) {
-            const auto result = laser_pose_in_base * Sophus::Vector3d{p.x(), p.y(), 0};
-            return std::make_pair(result.x(), result.y());
-          }) |
-          ranges::to<std::vector>,
-      force_filter_update_);
-  force_filter_update_ = false;
+      });
   const auto update_stop_time = std::chrono::high_resolution_clock::now();
   const auto update_duration = update_stop_time - update_start_time;
 
-  if (filter_updated) {
-    NODELET_INFO_THROTTLE(
-        2, "Particle filter update iteration stats: %ld particles %ld points - %.3fms",
-        particle_filter_->particle_count(), laser_scan->ranges.size(),
+  if (new_estimate.has_value()) {
+    last_known_estimate_ = new_estimate;
+
+    NODELET_INFO(
+        "Particle filter update iteration stats: %ld particles %ld points - %.3fms",
+        particle_filter_->particles().size(),  //
+        laser_scan->ranges.size(),             //
         std::chrono::duration<double, std::milli>(update_duration).count());
   }
 
-  // force publication of the first message and any subsequent updates to the filter
-  const auto publish_updated_estimations = !last_known_estimate_ || filter_updated;
-  if (publish_updated_estimations) {
-    last_known_estimate_ = particle_filter_->estimate();
+  if (!last_known_estimate_.has_value()) {
+    NODELET_WARN_THROTTLE(2, "Estimate not available for publishing");
+    return;
   }
 
-  const auto& [pose, covariance] = last_known_estimate_.value();
+  const auto& [base_pose_in_map, base_pose_covariance] = last_known_estimate_.value();
 
-  // new pose messages are only published on updates to the filter
-  if (publish_updated_estimations) {
+  // New pose messages are only published on updates to the filter.
+  if (new_estimate.has_value()) {
     auto message = geometry_msgs::PoseWithCovarianceStamped{};
     message.header.stamp = laser_scan->header.stamp;
     message.header.frame_id = config_.global_frame_id;
-    tf2::toMsg(pose, message.pose.pose);
-    tf2::covarianceEigenToRowMajor(covariance, message.pose.covariance);
+    tf2::toMsg(base_pose_in_map, message.pose.pose);
+    tf2::covarianceEigenToRowMajor(base_pose_covariance, message.pose.covariance);
     pose_pub_.publish(message);
-
-    // Update the estimation fo the transform between the global frame and the odom frame
-    latest_map_to_odom_transform_ = pose * base_pose_in_odom.inverse();
   }
 
-  // tranforms are always published to keep them current
-  if (latest_map_to_odom_transform_ && enable_tf_broadcast_ && config_.tf_broadcast) {
+  // Transforms are always published to keep them current
+  if (enable_tf_broadcast_ && config_.tf_broadcast) {
+    auto odom_transform_in_map = base_pose_in_map * base_pose_in_odom.inverse();
```

Comment:
@nahueespinosa nit: 
```suggestion
    const auto odom_transform_in_map = base_pose_in_map * base_pose_in_odom.inverse();
```

Reply:
Fixed in https://github.com/Ekumen-OS/beluga/pull/328/commits/0cefddb2d73f4dd0037bfa2ef75fdffc8a84d61a.